webui/react/src/components/TrialActions.tsx

```diff
@@ -0,0 +1,72 @@
+import { Button } from 'antd';
+import React, { useCallback, useState } from 'react';
+
+import { ButtonLoadingStates, ConditionalButton } from 'components/ExperimentActions';
+import Link from 'components/Link';
+import { launchTensorboard } from 'services/api';
+import { RunState, TBSourceType, TrialDetails } from 'types';
+import { openCommand } from 'utils/routes';
+import { terminalRunStates } from 'utils/types';
+
+import css from './TrialActions.module.scss';
+
+interface Props {
+  trial: TrialDetails;
+  onSettled: () => void; // A callback to trigger after an action is done.
+}
+
+enum Action {
+  Logs = 'Logs',
+  Continue = 'Continue',
+  Tensorboard = 'Tensorboard',
+}
+
+const TrialActions: React.FC<Props> = ({ trial, onSettled: updateFn }: Props) => {
+
+  const [ buttonStates, setButtonStates ] = useState<ButtonLoadingStates<Action>>({
+    Continue: false,
+    Logs: false,
+    Tensorboard: false,
+  });
+
+  const handleLaunchTensorboard = useCallback(() => {
+    setButtonStates(state => ({ ...state, tensorboard: true }));
+    launchTensorboard({ ids: [ trial.id ], type: TBSourceType.Trial })
+      .then((tensorboard) => {
+        openCommand(tensorboard);
+        return updateFn();
+      })
+      .finally(() => setButtonStates(state => ({ ...state, tensorboard: false })));
+  }, [ trial.id, updateFn ]);
+
+  const trialWillNeverHaveData = (trial: TrialDetails): boolean => {
+    const isTerminal = terminalRunStates.has(trial.state);
+    const stepsWithSomeMetric = trial.steps.filter(step => step.state === RunState.Completed);
+    return isTerminal && stepsWithSomeMetric.length === 0;
+  };
+
+  const actionButtons: ConditionalButton<TrialDetails>[] = [
+    { btn: <Button disabled key={Action.Continue} type="primary">Continue Trial</Button> },
+    { btn: <Button key={Action.Logs} type="primary">
+      <Link path={`/det/trials/${trial.id}/logs`}>Logs</Link>
+    </Button> },
+    {
+      btn: <Button key={Action.Tensorboard}
+        loading={buttonStates.Tensorboard} type="primary" onClick={handleLaunchTensorboard}>
+      Tensorboard</Button>,
+      showIf: (t): boolean => !trialWillNeverHaveData(t),
```

Comment:
**nit:** `trial` instead of `t`

Reply:
that would share the same name as the higher scope `trial`. `aTrial` for argument trial?

Reply:
https://github.com/determined-ai/determined/pull/931/commits/7607f855be8dfc12528ddbb6e7a0c2a8942373bd

Reply:
ah yeah `trial` under `props`... `targetTrial`?